### PR TITLE
[shopsys] added quick search in promo codes

### PR DESCRIPTION
**Pull request by grossmannmartin**
| Q             | A
| ------------- | ---
|Description, reason for the PR| Searching in the promo code list was never been easier.
|New feature| Yes 
|[BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/)| Yes 
|Fixes issues| ... 
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

----
:globe_with_meridians: Live Preview:
  - https://mg-search-promo-code.odin.shopsys.cloud
  - https://cz.mg-search-promo-code.odin.shopsys.cloud

### CODE REVIEW

File: project-base/app/src/Model/Order/PromoCode/Grid/PromoCodeGridFactory.php [hunk 1]
```diff
@@ -24,29 +24,38 @@ class PromoCodeGridFactory extends BasePromoCodeGridFactory
...
         parent::__construct($em, $gridFactory);
     }
 
     /**
      * @param bool $withEditButton
+     * @param string|null $search
      * @return \Shopsys\FrameworkBundle\Component\Grid\Grid
      */
-    public function create($withEditButton = true)
+    public function create($withEditButton = true, ?string $search = null)
```
Comment (TomasLudvik) [inline project-base/app/src/Model/Order/PromoCode/Grid/PromoCodeGridFactory.php:38]:
```suggestion
    public function create(bool $withEditButton = true, ?string $search = null)
```

File: project-base/app/templates/Admin/Content/PromoCode/list.html.twig [hunk 2]
```diff
@@ -2,6 +2,32 @@
...
+        <div class="in-tab__content">
+            <div class="in-tab__content__in active" data-tab-id="quick">
+
+                {{ form_start(quickSearchForm) }}
+                    <div class="box-quick-search">
+                        <div class="box-quick-search__input">
+                            {{ form_widget(quickSearchForm.text) }}
+                            <i class="svg svg-info cursor-help js-tooltip box-quick-search__icon in-icon in-icon--info"
+                               data-toggle="tooltip" data-placement="top"
+                               title="{{ 'Searching searches in the promo code codes.'|trans }}"
```
Comment (TomasLudvik) [inline project-base/app/templates/Admin/Content/PromoCode/list.html.twig]:
```suggestion
                               title="{{ 'Search searches in the codes of the promo codes.'|trans }}"
```

I would maybe removed this icon completely

File: project-base/app/templates/Admin/Content/PromoCode/list.html.twig [hunk 3]
```diff
@@ -2,6 +2,32 @@
...
+                {{ 'Quick search'|trans }}
+            </a>
+        </div>
+        <div class="in-tab__content">
+            <div class="in-tab__content__in active" data-tab-id="quick">
+
+                {{ form_start(quickSearchForm) }}
+                    <div class="box-quick-search">
+                        <div class="box-quick-search__input">
+                            {{ form_widget(quickSearchForm.text) }}
```
Comment (TomasLudvik) [inline project-base/app/templates/Admin/Content/PromoCode/list.html.twig]:
```suggestion
                            {{ form_widget(quickSearchForm.text, { attr:
                {
                    placeholder: 'Search by code'|trans
                }
            }) }}
```

File: upgrade/UPGRADE-v13.0.0-dev.md [hunk 4]
```diff
@@ -51,3 +51,10 @@ There you can find links to upgrade notes for other versions too.
     - already existing articles with this placement will be migrated to `Shopsys\FrameworkBundle\Model\Article\Article::PLACEMENT_NONE` placement
         - if you intend to keep using this placement, you can skip DB Migration `Shopsys\FrameworkBundle\Migrations\Version20230907132822` 
     - see #project-base-diff to update your project
+- allow quick searching in promo codes ([#2786](https://github.com/shopsys/shopsys/pull/2786))
+    - method `Shopsys\FrameworkBundle\Controller\Admin\PromoCodeController::listAction` changed its interface:
```
Comment (TomasLudvik) [inline upgrade/UPGRADE-v13.0.0-dev.md:100]: Should we also mention the change of PromoCodeGridFactory::create?

**Review by TomasLudvik — CHANGES_REQUESTED** [review]

File: project-base/app/src/Model/Order/PromoCode/Grid/PromoCodeGridFactory.php (hunk 1 above)
Comment (grossmannmartin) [reply project-base/app/src/Model/Order/PromoCode/Grid/PromoCodeGridFactory.php:38]: that would require the  BC Break change of the framework's PromoCodeGridFactory

File: upgrade/UPGRADE-v13.0.0-dev.md (hunk 4 above)
Comment (grossmannmartin) [reply upgrade/UPGRADE-v13.0.0-dev.md:100]: PromoCodeGridFactory::create in the framework package is unchanged and the project-base implementation is covered by #project-base-diff

File: project-base/app/templates/Admin/Content/PromoCode/list.html.twig (hunk 3 above)
Comment (grossmannmartin) [reply project-base/app/templates/Admin/Content/PromoCode/list.html.twig]: added

File: project-base/app/templates/Admin/Content/PromoCode/list.html.twig (hunk 2 above)
Comment (grossmannmartin) [reply project-base/app/templates/Admin/Content/PromoCode/list.html.twig]: I have removed the icon in favor of the placeholder (below)

**Review by TomasLudvik — APPROVED** [review]

**Pull request merged**